### ISSUE TITLE
choosenim installs old version of mingw causing nim cpp to be broken on windows

### ISSUE THREAD

**Issue by rayman22201** [dom96/choosenim#129]
Problem:
on Windows, (I am using windows 10), choosenim installs Mingw 4.9.1, but the latest devel recommends Mingw 6.3.0

Using the old version of Mingw results in `nim cpp` producing broken exe's.
Nim will compile your program, but running it produces no output. Even a simple hello world program does not work.

Manually downloading and installing Mingw 6.3.0 from the nim website, and forcing choosenim to use that version fixes the problem.

Steps:
1. Clear out out the `C:\User\<username>\.nimble` and `C:\User\<username>\.choosenim` directories first
2. Download the latest windows release (v 0.4.0) from the github release page
3. Extract and run `runme.bat`
4. Open Powershell 
5. run `choosenim update "#head"`
6. Run `nim cpp -r hello_world.nim` -> this fails

Steps to force Mingw 6.3.0
1. remove `g++.exe` and `gcc.exe` from `C:\User\<username>\.nimble`
(I used Get-Command to determine which g++ was being used. This is the powershell equivalent of the Unix `which` command)
2. Download Mingw 6.3.0 from https://nim-lang.org/install_windows.html
3. Extract the zip file
4. Open Powershell
4. Add the Mingw directory to your windows path.
In Powershell I did the following to temporarily add to my path: `$env:Path += "C:\Users\ray\Downloads\mingw64-6.3.0\mingw64\bin"`
5. run `choosenim update "#head"`
6. Run `nim cpp -r hello_world.nim` -> this now succeeds

**Comment by rayman22201** [comment]
After talking to @genotrance on irc:
This may be related to me having previously having a very old version of choosenim and nim installed. It's possible that the download caching mechanism picked up the old mingw zip file, and did not downlad the newer mingw.

This emphasizes to me the need for #130. choosenim should verify the mingw version on windows.

**Comment by rayman22201** [comment]
Just figured it out. The hard coded link here: https://github.com/dom96/choosenim/blob/master/src/choosenim/download.nim#L17

Still points to the 4.9.1 version of mingw.
It should be updated to: https://nim-lang.org/download/mingw32-6.3.0.7z

**Comment by akavel** [comment]
Currently, this resulted in zip/zipfiles failing to compile for me, with a cryptic error message:

```
        ... In file included from c:/users/mateusz/.choosenim/toolchains/mingw32/i686-w64-mingw32/include/zconf.h:452:0,
        ...                  from c:/users/mateusz/.choosenim/toolchains/mingw32/i686-w64-mingw32/include/zlib.h:34,
        ...                  from C:\Users\Mateusz\.nimble\pkgs\zip-0.2.1\zip\private\libzip_all.c:34:
        ... c:/users/mateusz/.choosenim/toolchains/mingw32/i686-w64-mingw32/include/unistd.h:12:20: fatal error: getopt.h: No such file or directory
        ...  #include <getopt.h>
        ...                     ^
        ... compilation terminated.
```

Through some lucky intuition and coincidence, I managed to guess that this may be old version of MinGW (which I confirmed with `gcc --version` to be now *super ancient* 4.9.1, with Nim 1.0.0 recommending gcc 6.3.0). I upgraded the choosenim's gcc with the following steps:

1. Delete `%userprofile%\.choosenim\toolchains\mingw32` folder
2. Download https://nim-lang.org/download/mingw32-6.3.0.7z
3. Unpack it into `%userprofile%\.choosenim\toolchains\`
4. `gcc --version` now prints 6.3.0

nimble now compiles my project using `zip/zipfiles` - though it fails at linking stage, without libzip.dll, but that's a completely different story...

Could choosenim be at least temporarily updated to download mingw6.3.0 (until the libarchive thing gets done), given that this is what is used by the finally released Nim 1.0.0? This could hopefully improve first experience of Nim for any people trying it for the first time with choosenim...
